linkerd/app/src/identity.rs

```diff
@@ -45,6 +47,18 @@ impl Config {
         match self {
             Config::Disabled => Ok(Identity::Disabled),
             Config::Enabled { control, certify } => {
+                const EWMA_DEFAULT_RTT: Duration = Duration::from_millis(30);
+                const EWMA_DECAY: Duration = Duration::from_secs(10);
+                let discover = {
+                    const BUFFER_CAPACITY: usize = 1_000;
+                    let cache_timeout = Duration::from_secs(60);
+                    discover::Layer::new(
+                        BUFFER_CAPACITY,
+                        cache_timeout,
+                        control::dns_resolve::Resolve::new(dns),
+                    )
```

Comment:
So, while I think the larger change looks fine, I'm a bit unsure on this detail... specifically how the watchdog timeout (what you call `cache_timeout` here) will interact with these clients.

The watchdog timeout probably isn't even needed anymore, but it was there to detect the case when the client had been orphaned such that discovery updates would back up in the buffer, and then eventually exert backpressure on the client. But in cases like the identity client, we may only send something like one request a day; so in this case we _expect_ the client to be idle (and not dropped) for long periods of time. So I worry that the watchdog timeout is inappropriate here.

Reply:
On further thought, I don't think we even _need_ a buffer around these resolutions?

In the discovery case, resolutions are sent by the server to the proxy and we have to consume them all to prevent the stream/window from backing up. In the DNS case, everything is driven by the act of polling the stream. So I don't think we have anything to buffer, really.